handlers/utils/utils.go

```diff
@@ -141,11 +118,17 @@ func readBuffer(reader io.ReadCloser) string {
 }
 
 func isStrSet(m map[string]interface{}, key string) bool {
-	return m[key] != nil && len(m[key].([]string)) > 0
+	ok := false
+	switch m[key].(type) {
+	case string:
+		ok = len(InterfaceToString(m[key])) > 0
+	case []string:
+		ok = len(m[key].([]string)) > 0
+	}
+	return m[key] != nil && ok
 }
 
-// this method check if a field exists in a map
-func getFieldsIfExist(m map[string]interface{}, fields ...string) (interface{}, bool) {
+func GetFieldsIfExist(m map[string]interface{}, fields ...string) (interface{}, bool) {
```

Comment:
I cant comment on it because it is hidden in this diff, but in this logic:
```
default:
 			// if it is the last field and it is not empty
 			// it exists othewise return false
 			if i == len(fields)-1 {
 				return tempMap[field], true
 			}
 			return nil, false
 		}
```
return tempMap[field], true is wrong because tempMap may not actually have field as key. I guess you'd want something like:
```
val, ok := tempMap[field]
return val, ok
```